backend/tool_instance_v2/serializers.py

```diff
@@ -51,8 +55,102 @@ def to_representation(self, instance: ToolInstance) -> dict[str, str]:
             return rep
         rep[ToolKey.ICON] = tool.icon
         rep[ToolKey.NAME] = tool.properties.display_name
+
+        # Transform adapter IDs back to names for UI display
+        metadata = rep.get(TIKey.METADATA, {})
+        if metadata:
+            rep[TIKey.METADATA] = self._transform_adapter_ids_to_names_for_display(
+                metadata, tool_function
+            )
+
         return rep
 
+    def _transform_adapter_ids_to_names_for_display(
+        self, metadata: dict[str, Any], tool_uid: str
+    ) -> dict[str, Any]:
+        """Transform adapter IDs back to names for UI display."""
+        # Create a copy to avoid mutating the original metadata
+        display_metadata = metadata.copy()
+
+        try:
+            tool: Tool = ToolProcessor.get_tool_by_uid(tool_uid=tool_uid)
+            schema = ToolUtils.get_json_schema_for_tool(tool)
+
+            # Create mapping of adapter keys to their types
+            adapter_key_to_type = {}
+            for key in schema.get_llm_adapter_properties().keys():
+                adapter_key_to_type[key] = AdapterTypes.LLM
+            for key in schema.get_embedding_adapter_properties().keys():
+                adapter_key_to_type[key] = AdapterTypes.EMBEDDING
+            for key in schema.get_vector_db_adapter_properties().keys():
+                adapter_key_to_type[key] = AdapterTypes.VECTOR_DB
+            for key in schema.get_text_extractor_adapter_properties().keys():
+                adapter_key_to_type[key] = AdapterTypes.X2TEXT
+            for key in schema.get_ocr_adapter_properties().keys():
+                adapter_key_to_type[key] = AdapterTypes.OCR
+
+            # Transform IDs back to names for display
+            for adapter_key, adapter_type in adapter_key_to_type.items():
+                if adapter_key in display_metadata and display_metadata[adapter_key]:
+                    adapter_value = display_metadata[adapter_key]
+                    # If it's a UUID (adapter ID), convert to name for display
+                    if ToolInstanceHelper.is_uuid_format(adapter_value):
+                        try:
+                            adapter_name = AdapterProcessor.get_adapter_instance_by_id(
+                                adapter_value
+                            )
```

Comment:
```suggestion
                            adapter_name = AdapterInstance.objects.get(id=adapter_instance_id).adapter_name
```
Directly use this, `AdapterProcessor.get_adapter_instance_by_id()` is error prone since it suppresses exceptions and then attempts to access the adapter name. If possible correct it